libraries/chain/controller.cpp

```diff
@@ -1512,7 +1514,14 @@ struct controller_impl {
                // irreversible. Instead, this moves irreversible as much as possible and allows the next maybe_switch_forks call to apply these
                // non-validated blocks. After the maybe_switch_forks call (before next produced block or on next received block), irreversible
                // can then move forward on the then validated blocks.
-               return read_mode == db_read_mode::IRREVERSIBLE || bsp->is_valid();
+               // In irreversible mode, break every ~500ms to allow other tasks (e.g. get_info, SHiP) opportunity to run. There is a post
+               // for every incoming blocks; enough posted tasks to apply all blocks queued to the fork db.
+               if (read_mode == db_read_mode::IRREVERSIBLE) {
+                  if (!replaying && fc::time_point::now() - start > fc::milliseconds(500))
```

Comment:
Shouldn't `start` be reset?

Reply:
Once this returns `false`, the method exits.